llvm/include/llvm/Frontend/HLSL/HLSLBinding.h

```diff
@@ -152,6 +172,12 @@ class BindingInfoBuilder {
         [&HasOverlap](auto, auto) { HasOverlap = true; });
   }
 
+  LLVM_ABI BoundRegs getBoundRegs() {
```

Comment:
I think `takeBoundRegs` would be a clearer name, since this modifies/invalidates the `BindingInfoBuilder`